packages/shared/lib/core/wallet/actions/mintNativeToken.ts

```diff
@@ -0,0 +1,47 @@
+import { selectedAccount } from '@core/account'
+import { NativeTokenOptions, TransactionOptions } from '@iota/wallet'
+import { Converter } from '@lib/converter'
+import { isTransferring } from '@lib/wallet'
+import { get } from 'svelte/store'
+import { Activity } from '../classes'
+import { VerificationStatus } from '../enums'
+import { buildPersistedAssetFromIrc30Metadata } from '../helpers'
+import { IIrc30Metadata, IPersistedAsset } from '../interfaces'
+import { addActivityToAccountActivitiesInAllAccountActivities } from '../stores'
+import { addPersistedAsset } from '../stores/persisted-assets.store'
+
+export async function mintNativeToken(
+    maximumSupply: number,
+    circulatingSupply: number,
+    metadata: IIrc30Metadata
+): Promise<void> {
+    try {
+        isTransferring.set(true)
+        const account = get(selectedAccount)
+        const nativeTokenOptions: NativeTokenOptions = {
+            accountAddress: account.depositAddress,
+            maximumSupply: '0x' + maximumSupply.toString(16),
+            circulatingSupply: '0x' + circulatingSupply.toString(16),
+            foundryMetadata: Array.from(Converter.utf8ToBytes(JSON.stringify(metadata))),
```

Comment:
this is required to be a hexstring for the new wallet.rs version

Reply:
Already updated